units/src/weight.rs

```diff
@@ -123,7 +123,7 @@ impl Weight {
     /// Computes `self / rhs` returning `None` if `rhs == 0`.
     pub fn checked_div(self, rhs: u64) -> Option<Self> { self.0.checked_div(rhs).map(Self) }
 
-    /// Scale by witness factor.
+    /// Scales by the witness factor.
     ///
     /// Computes `self * WITNESS_SCALE_FACTOR` returning `None` if an overflow occurred.
     pub fn scale_by_witness_factor(self) -> Option<Self> {
```

Comment:
Why the hell do we have this function here? Weight already accounts for witness, so this is wrong. It should've taken `usize` and be called `from_non_discounted_size` (and also have `from_discounted_size`).

Reply:
@Kixunil its not clear to me from your comment what the priority is with this problem. If you really care about something but don't have the time to go further can you mention me please and ask me to investigate it? (I think its fine to make requests of my time because I'm paid to be here, for others I don't think we should do it too much). Cheers man. (And well handled Jamil, we are a prickly bunch round here, we have a tendency to get a bit emotionally involved in our work, don't take it personally.)

Reply:
@tcharding thanks! Yes, that'd be appreciated. The priority is: I don't want to stabilize `units` with this function.

Reply:
No worries, added #2966